pkg/connector/role.go

```diff
@@ -293,7 +295,7 @@ func (r *roleResourceType) Grant(ctx context.Context, principal *v2.Resource, en
 		},
 	)
 	if err != nil {
-		return nil, fmt.Errorf("crowdstrike-connector: failed to grant role membership: %w", err)
+		return nil, wrapCrowdStrikeError(err, "grant: failed to assign role membership")
```

Comment:
Do we need to check 409 conflict (or whatever crowdstrike returns when you try to grant an already-created grant) and return no error along with the GrantAlreadyExists annotation? The same goes for revoking and the GrantAlreadyRevoked annotation.